src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/RequireEmptyLineBeforeAtClauseBlockCheck.java

```diff
@@ -0,0 +1,197 @@
+ *  </p>
+ *  <pre>
+ *  &#47;**
+ *  * testMethod's javadoc.
+ *  *
+ *  * &#64;param firstParam
+ *  * &#64;return something
+ *  *&#47;
+ *  public boolean testMethod(int firstParam) {
+ *      return false;
+ *  }
+ *  </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.tag.line.before}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 8.35
+ */
+@StatelessCheck
+public class RequireEmptyLineBeforeAtClauseBlockCheck extends AbstractJavadocCheck {
+
+    /**
+     * The key in "messages.properties" for the message that describes an at-clause in javadoc
+     * requiring an empty line before it.
+     */
+    public static final String MSG_JAVADOC_TAG_LINE_BEFORE = "javadoc.tag.line.before";
+
+    /**
+     * When javadoc has only a javadoc tag like {@literal @} in it, the JAVADOC_TAG in a JAVADOC
+     * detail node will always have 3 siblings before it. The parse tree looks like:
+     * <pre>
+     * JAVADOC
+     * \
+     *  -- NEWLINE
+     *  -- LEADING_ASTERISK
+     *  -- WS
+     *  -- JAVADOC_TAG
+     * </pre>
+     */
+    private static final int NUM_OF_SIBLINGS_BEFORE_ONLY_JAVADOC_TAG = 3;
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC_TAG,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    /**
+     * Logs when there is no empty line before the at clause.
+     *
+     * @param atClause the at at-clause node to check for an empty space before it.
+     */
+    @Override
+    public void visitJavadocToken(DetailNode atClause) {
+        // No need to filter token because overridden getDefaultJavadocTokens ensures that we only
+        // receive JAVADOC_TAG DetailNode.
+        if (!isAnotherAtClauseBefore(atClause)
+                && !isOnlyAtClauseNothingElse(atClause)
+                && insufficientConsecutiveNewlines(atClause)) {
+            log(atClause.getLineNumber(),
+                    MSG_JAVADOC_TAG_LINE_BEFORE,
+                    atClause.getChildren()[0].getText());
+        }
+    }
+
+    private static boolean isAnotherAtClauseBefore(DetailNode atClause) {
+        boolean found = false;
+        DetailNode currentClause = JavadocUtil.getPreviousSibling(atClause);
+        while (currentClause != null) {
+            if (currentClause.getType() == JavadocTokenTypes.JAVADOC_TAG) {
+                found = true;
+                break;
+            }
+            currentClause = JavadocUtil.getPreviousSibling(currentClause);
+        }
+        return found;
+    }
+
+    private static boolean isOnlyAtClauseNothingElse(DetailNode atClause) {
```

Comment:
`isOnlyAtClauseInWholeJavadoc`

Reply:
done. also renamed AtClause to Tag based on below comment